.github/workflows/tests.yml

```diff
@@ -71,33 +71,44 @@ jobs:
       matrix:
         sanitiser: [None, Address, Thread, Undefined]
     env:
+      CONAN_CACHE_MOUNT_SOURCE: ~/.conan/
       HOST_TYPE: ci
-      REDIS_QUEUE_HOST: redis
-      REDIS_STATE_HOST: redis
-      ASAN_OPTIONS: "verbosity=1:halt_on_error=1"
+      LOG_LEVEL: info
+      # Overwritting the entrypoint makes sure that we use the latest version
+      # of the planner
+      PLANNER_ENTRYPOINT_BINARY: /code/faabric/build/static/bin
+      ASAN_OPTIONS: verbosity=1:halt_on_error=1
       TSAN_OPTIONS: "verbosity=1 halt_on_error=1 suppressions=./thread-sanitizer-ignorelist.txt history_size=4"
-      UBSAN_OPTIONS: "print_stacktrace=1:halt_on_error=1"
-    container:
-      image: faasm.azurecr.io/faabric:0.4.2
-      credentials:
-        username: ${{ secrets.ACR_SERVICE_PRINCIPAL_ID }}
-        password: ${{ secrets.ACR_SERVICE_PRINCIPAL_PASSWORD }}
-      options: --privileged
-    services:
-      redis:
-        image: redis
+      UBSAN_OPTIONS: print_stacktrace=1:halt_on_error=1
     steps:
+      - name: "Install Conan"
```

Comment:
I move the `tests` job from running inside a container with additional services to using `docker compose` manually.

The `planner` runs in a separate container, but may be updated frequently. To avoid having to re-tag the code everytime we change anything in the planner, we want to mount the binaries we build as part of the tests (with the latest checkout of the code) and use those to run the planner.

The first problem this arises is that, even though GHA's service containers support mounting volumes, the planner service requires the `planner_server` binary to be there before it can start. And once it is running it can not be restarted from _inside_ the tests container.

To overcome this problem, we add a custom entrypoint for the planner that allows setting the directory where it will look for the `planner_server` binary. If its not there, it will wait until it becomes available.

Then, we come to a second problem. GHA's service containers don't allow overwriting the entrypoint. See [here](https://github.com/orgs/community/discussions/26688) for a further discussion.

The solution I have found is to use `docker compose` manually.